### ISSUE TITLE
Webinterface show empty values

### ISSUE THREAD

**Issue by IMmmKI** [Newcomer1989/TSN-Ranksystem#398]
I @Shad86 linked original thread hear https://github.com/Newcomer1989/TSN-Ranksystem/issues/408 for better handly on this thread (other issue / behavior)

**Comment by Newcomer1989** [comment]
The first screenshot shows you the problem. You have a wrong rankup definition. That is the reason the Ranksystem doesn't startup correct.
The first problem initiate the second one. Your files version is 1.2.4 but your database version is 1.2.3.

So you need to correct your rankup defintion to solve all your problems.

If I am right, you will now not be able to save it via webinterface.
You need to update this value direct on the database. Enter the following SQL command to correct it:

`UPDATE config SET grouptime='31534000=>XXX';`
Replace the XXX with an valid server group ID on your ts3 server.

After this change you should be able to start the Bot. Then the update process should update your database to version 1.2.4., which should be shown after also on the Webinterface (reload needed).

Then you are fine!

**Comment by IMmmKI** [comment]
Ahh, Awesome! Thank you so much @Newcomer1989 . 

That has solved the problem for me. 

Thanks! 🥇

**Comment by Newcomer1989** [comment]
Perfect :)

**Comment by Ultrashadow** [comment]
![grafik](https://user-images.githubusercontent.com/24988303/36175852-13099218-1111-11e8-95d1-7ce510e899af.png)
Hey, I think I got a similar problem here, but the funny thing is that those groups in the error message aren't defined in server group exception. Even more strange that the setting page is empty while having all data stored unchanged in the db. 
![grafik](https://user-images.githubusercontent.com/24988303/36175979-79a23f3e-1111-11e8-8b5c-cb09badc05aa.png)
Even if you edit those settings, nothing happens. I tried that db solution and I renewed the files already. Overall, the ranksystem works and also uses the rankup definition but it does not appear on the setting page :/

**Comment by Newcomer1989** [comment]
You have an other problem as this was starting with the thread. The described update to the DB is in your case wrong.
I hope, you have a backup of your rankup definition or it is gone now!

Can you do a select on the table config and show us the result here?
`SELECT * FROM config;`
Please remove passwords before posting!

**Comment by Ultrashadow** [comment]
Well, I downloaded a backup yesterday, but I got the definition of rankups anyway since I calculate them with a formular. I needed to kill the ranksystem cause it was about to give users admin ranks, lel.


`user	pass	ip	10011	9986	user	pass	en	Ranksystem	RankSystem	31534000=>12	1	1	1	uid	7200	01.02.2004	1	uid	2,9,111,80,85,144,107,114,176	%a days, %h hours, %i mins, %s secs	0	1	0	0	1	0	1	1	1	1	1	0	1	86400	0		1	0	Europe/Berlin	/directory	0	0	1483564302	600		msg	0	01.02.2004	news	uid	1	info	info	info	1	0	0	version	0	17219
`

**Comment by Newcomer1989** [comment]
Can we get it in a better view, that we are able to separat one column from another one?
Also the headline would be very nice. The config table get changed with every version, so it is easier to detect a missed column.

**Comment by Ultrashadow** [comment]
![grafik](https://user-images.githubusercontent.com/24988303/36221946-0aefda60-11c0-11e8-9429-80b93e617719.png)

Maybe thats more helpful

**Comment by FreakyFP** [comment]
Habe so ein ähnliches problem nachdem ich den ts3server auf 3.1 geupdatet habe, angeblich hätten sich die Servergroup ids geändert das ist aber nicht der Fall...
Habe es über SGL l wie oben gesagt versucht, der Fehler besteht weiterhin

```
[2018-02-15 13:37:22.784085   CRITICAL  Critical Config error! Shutting down!
2018-02-15 13:37:22.780166   ERROR         There is a servergroup with ID 113 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.780113   ERROR         There is a servergroup with ID 191 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.780061   ERROR         There is a servergroup with ID 114 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.780005   ERROR         There is a servergroup with ID 8 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779930   ERROR         There is a servergroup with ID 100 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779828   ERROR         There is a servergroup with ID 99 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779741   ERROR         There is a servergroup with ID 98 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779656   ERROR         There is a servergroup with ID 97 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779563   ERROR         There is a servergroup with ID 96 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779470   ERROR         There is a servergroup with ID 95 configured inside your 'servergroup exception' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779347   CRITICAL      There is a servergroup with ID 219 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779266   CRITICAL      There is a servergroup with ID 218 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779186   CRITICAL      There is a servergroup with ID 217 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779101   CRITICAL      There is a servergroup with ID 216 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.779012   CRITICAL      There is a servergroup with ID 215 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778885   CRITICAL      There is a servergroup with ID 214 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778803   CRITICAL      There is a servergroup with ID 213 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778724   CRITICAL      There is a servergroup with ID 212 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778646   CRITICAL      There is a servergroup with ID 211 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778567   CRITICAL      There is a servergroup with ID 210 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778467   CRITICAL      There is a servergroup with ID 209 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778374   CRITICAL      There is a servergroup with ID 208 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.778187   CRITICAL      There is a servergroup with ID 207 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.770310   CRITICAL      There is a servergroup with ID 206 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.770217   CRITICAL      There is a servergroup with ID 205 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.770135   CRITICAL      There is a servergroup with ID 204 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.770065   CRITICAL      There is a servergroup with ID 203 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769985   CRITICAL      There is a servergroup with ID 202 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769928   CRITICAL      There is a servergroup with ID 201 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769860   CRITICAL      There is a servergroup with ID 200 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769781   CRITICAL      There is a servergroup with ID 199 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769724   CRITICAL      There is a servergroup with ID 198 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769668   CRITICAL      There is a servergroup with ID 197 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769610   CRITICAL      There is a servergroup with ID 196 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769522   CRITICAL      There is a servergroup with ID 195 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.769395   CRITICAL      There is a servergroup with ID 193 configured inside your 'rank up definition' paramter (webinterface -> core), but that servergroup ID isn't existent on your TS3 server (anymore)! Please correct this or errors might happen!
2018-02-15 13:37:22.768557   INFO        Config check started...
2018-02-15 13:37:22.768502   DEBUG     Using PHP Version: 7.0.25-0ubuntu0.16.04.1
2018-02-15 13:37:22.768378   DEBUG     Running on OS: Linux 4.4.0-96-generic
2018-02-15 13:37:22.760325   INFO      Initialize Bot...
```

**Comment by FreakyFP** [comment]
Zum testen habe ich ein komplett neues ranksystem installiert und bekomme diesen error wenn ich die core Einstellungen speichern will.

`Array ( [0] => 42S22 [1] => 1054 [2] => Unknown column 'registercid' in 'field list' )`

**Comment by GYJohn** [comment]
Is it possible to deactive this the next rank up?
I only need the stats and the next rank field can't be empty.

**Comment by Newcomer1989** [comment]
I will ignore other questions in this post in order to maintain an overview.
@GYJohn & @FreakyFP please open an own issue for your questions. I think it has nothing to do with the issue of UltraShadow.

@Ultrashadow 
The columns currvers and newversion hast wrong values. There should stand "1.2.4" inside. It is only the post here or are this values also inside your database?
If the looks in the database same like in your post, please correct this!

`UPDATE config SET currvers = '1.2.4', newversion = '1.2.4';`

**Comment by Ultrashadow** [comment]
Oh, I just noticed, that was excel formatting the field to a 'date' value, lol. In the database it is correct. Excel is doing that all the time, just M$ things

**Comment by Newcomer1989** [comment]
Ok, dann sollte die DB in Ordnung sein.

Bitte probiere einmal folgendes:
1) Stoppen des Bots und auch des automatischen Starts (Cron/Task)
2) Sicherung des Webspaces und der DB
3) Download der Version 1.2.4 von unserer Seite
4) Überspielen der Files auf dem Webspache mit denen aus dem Download; Nicht überschrieben werden sollten `other/dbconfig.php`, `other/phpcommand.php` und auch nicht hochgeladen werden sollte die `install.php`

Danach den Bot noch einmal regulär starten und das Webinterface überprüfen.

Falls das auch nicht helfen sollte, bitte die Tage auf unserem Public TS (TS3 Adresse: ts-n.net) melden. Dann müssten wir uns das Problem mal gemeinsam anschauen.

**Comment by Ultrashadow** [comment]
I can't exactly tell why it is working now since I already did copy & overwrite the files to the server, but after doing it two times again, it worked. Eventually, the error log showed up some permission issues, which I fixed. So maybe there was something with permissions issues but they were not displayed in the ranking logs so I did not recognize it. 
Alright, thanks for your patience :).

War auch tatsächlich nur ne Sekunde davon entfernt, auf eurem Server aufzuschlagen und dann gehts natürlich ._. - die Daten werden auch wieder richtig angezeigt. Danke euch :)
